### ISSUE TITLE
Aggregate hydrator EventManager via factory overwritten by Mvc\ServiceManagerConfig initializer

### ISSUE THREAD

**Issue by robob4him** [zendframework/zend-hydrator#40]
We have a specific strategy for mappers to hydrate entities using the AggregateHydrator and it's events. We have a hydrator factory which initializes an AggregateHydrator and adds a ClassMethods hydrator. In ZF3, because the AggregateHydrator implements the EventManagerAwareInterface and Mvc\ServiceManagerConfig explicitly adds an EventManagerAwareInitializer, AggregateHydrator's EventManager is explicitly overwritten post-factory and ClassMethods is lost.

To summarize:

``` php
ServiceManager
{
    doCreate
    {
        FactoryInterface
        {
            new AggregateHydrator
            AggregateHydrator->add(ClassMethods) // Creates new EventManager
        }()

        EventManagerAwareInitializer {
            AggregateHydrator->setEventManager($container->get('EventManager')) // No ClassMethods
        }
    }
}
```

**Comment by BreyndotEchse** [comment]
Provide the EventManager yourself first. Then `EventManagerAwareInitializer` will not overwrite the EventManager.
```php
ServiceManager
{
    doCreate
    {
        FactoryInterface
        {
            new AggregateHydrator
            AggregateHydrator->setEventManager($container->get('EventManager'))
            AggregateHydrator->add(ClassMethods) // Creates new EventManager
        }()

        EventManagerAwareInitializer {
            // Does nothing now
        }
    }
}
```

**Comment by froschdesign** [comment]
I will close this issue, because the question has been answered.
